packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2629,8 +2629,16 @@ class InputDecoration {
   /// Sets the [contentPadding] property to [EdgeInsets.zero].
   const InputDecoration.collapsed({
     required this.hintText,
-    this.floatingLabelBehavior,
-    this.floatingLabelAlignment,
+    @Deprecated(
+      'Invalid parameter because a collapsed decoration has no label. '
+      'This feature was deprecated after v3.24.0-0.1.pre.',
+    )
+    FloatingLabelBehavior? floatingLabelBehavior,
```

Comment:
Why couldn't this be `this.floatingLabelBehavior`?

Reply:
Because when adding @deprecated to `this.floatingLabelBehavior` the analyzer will ask for the field to be deprecated (see https://dart.dev/tools/linter-rules/deprecated_consistency, especially the third case below).

```
Details
[#](https://dart.dev/tools/linter-rules/deprecated_consistency#details)
DO apply @Deprecated() consistently:

- if a class is deprecated, its constructors should also be deprecated.
- if a field is deprecated, the constructor parameter pointing to it should also be deprecated.
- if a constructor parameter pointing to a field is deprecated, the field should also be deprecated.
```

This lint seems too restrictive as one would like to deprecate a parameter in one particular constructor but not in other constructors. This is my first deprecation so maybe I'm missing something? Or maybe the workaround I used is expected?

Reply:
I'm not sure if this case has come up yet, as in removing one parameter from one constructor of a class, but not the others. Looks like these parameters were doing nothing before with the `collapsed` constructor? We often don't directly assign parameters when deprecating them though, so I think this is fine.